x-pack/filebeat/input/http_endpoint/config.go

```diff
@@ -13,37 +13,41 @@ import (
 
 // Config contains information about httpjson configuration
 type config struct {
-	TLS           *tlscommon.ServerConfig `config:"ssl"`
-	BasicAuth     bool                    `config:"basic_auth"`
-	Username      string                  `config:"username"`
-	Password      string                  `config:"password"`
-	ResponseCode  int                     `config:"response_code" validate:"positive"`
-	ResponseBody  string                  `config:"response_body"`
-	ListenAddress string                  `config:"listen_address"`
-	ListenPort    string                  `config:"listen_port"`
-	URL           string                  `config:"url"`
-	Prefix        string                  `config:"prefix"`
-	ContentType   string                  `config:"content_type"`
+	TLS              *tlscommon.ServerConfig `config:"ssl"`
+	BasicAuth        bool                    `config:"basic_auth"`
+	Username         string                  `config:"username"`
+	Password         string                  `config:"password"`
+	ResponseCode     int                     `config:"response_code" validate:"positive"`
+	ResponseBody     string                  `config:"response_body"`
+	ListenAddress    string                  `config:"listen_address"`
+	ListenPort       string                  `config:"listen_port"`
+	URL              string                  `config:"url"`
+	Prefix           string                  `config:"prefix"`
+	ContentType      string                  `config:"content_type"`
+	AuthHeader       string                  `config:"auth_header"`
```

Comment:
wondering if it makes more sense to make these nested and named something like `secret`? so something like:

```yaml
secret:
  header: ...
  value: ...
```

If we wanted we could make the default value for `header` `Authorization` too, but the name `auth_header` seems almost a bit too tied to an `Authorization` header (think most webhook apis call them 'secrets' anyway)